src/main/java/com/metamx/common/io/smoosh/FileSmoosher.java

```diff
@@ -53,6 +55,7 @@
  * <p/>
  * It does not split input files among separate output files, instead the various "chunk" files will
  * be varying sizes and it is not possible to add a file of size greater than Integer.MAX_VALUE
+ * TODO(leventov): consider making this class final
```

Comment:
I know this isn't Druid, but generally in Druid we try to minimize TODOs in the code, in favor of Github issues or in favor of just actually doing the thing if it's really important.